src/main/java/io/gravitee/policy/rest2soap/RestToSoapTransformerPolicy.java

```diff
@@ -63,6 +63,8 @@ public void onRequest(Request request, Response response, ExecutionContext execu
                 !soapTransformerPolicyConfiguration.getSoapAction().trim().isEmpty()) {
             LOGGER.debug("Add a SOAPAction header to invoke SOAP WS: {}", soapTransformerPolicyConfiguration.getSoapAction());
             request.headers().set(SOAP_ACTION_HEADER, soapTransformerPolicyConfiguration.getSoapAction());
+	    String charset =  soapTransformerPolicyConfiguration.getCharset() ;
+            request.headers().set("content-type", (charset == null ? MediaType.TEXT_XML : MediaType.TEXT_XML + "; charset="+charset ) );
```

Comment:
Please use the constant value from `io.gravitee.common.http.HttpHeaders` for the content-type header.

also, please check that charset can be null or empty.